test/integration/gardenlet/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,324 @@
+			Data: map[string][]byte{
+				"foo": []byte("bar"),
+			},
+		}
+
+		Expect(testClient.Create(ctx, gardenSecret)).To(Succeed())
+		log.Info("Created Secret for BackupBucket in garden for test", "secret", client.ObjectKeyFromObject(gardenSecret))
+
+		DeferCleanup(func() {
+			By("deleting secret for BackupBucket in garden")
+			Expect(testClient.Delete(ctx, gardenSecret)).To(Succeed())
+		})
+
+		By("creating BackupBucket")
+		backupBucket = &gardencorev1beta1.BackupBucket{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "foo-",
+				Labels:       map[string]string{testID: testRunID},
+			},
+			Spec: gardencorev1beta1.BackupBucketSpec{
+				Provider: gardencorev1beta1.BackupBucketProvider{
+					Type:   "provider-type",
+					Region: "some-region",
+				},
+				ProviderConfig: &runtime.RawExtension{Raw: []byte(`{"dash":"baz"}`)},
+				SecretRef: corev1.SecretReference{
+					Name:      gardenSecret.Name,
+					Namespace: gardenSecret.Namespace,
+				},
+				SeedName: &seed.Name,
+			},
+		}
+
+		Expect(testClient.Create(ctx, backupBucket)).To(Succeed())
+		log.Info("Created BackupBucket for test", "backupBucket", client.ObjectKeyFromObject(backupBucket))
+
+		DeferCleanup(func() {
+			By("deleting BackupBucket")
+			Expect(testClient.Delete(ctx, backupBucket)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		extensionSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateBackupBucketSecretName(backupBucket.Name),
+				Namespace: gardenNamespace.Name,
+			},
+		}
+
+		extensionBackupBucket = &extensionsv1alpha1.BackupBucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: backupBucket.Name,
+			},
+		}
+
+		coreGeneratedSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateGeneratedBackupBucketSecretName(backupBucket.Name),
+				Namespace: gardenNamespace.Name,
+			},
+		}
+
+		expectedExtensionBackupBucketSpec = extensionsv1alpha1.BackupBucketSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type:           backupBucket.Spec.Provider.Type,
+				ProviderConfig: backupBucket.Spec.ProviderConfig,
+			},
+			Region: backupBucket.Spec.Provider.Region,
+			SecretRef: corev1.SecretReference{
+				Name:      extensionSecret.Name,
+				Namespace: extensionSecret.Namespace,
+			},
+		}
+
+		By("creating generated secret in seed")
+		seedGeneratedSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateGeneratedBackupBucketSecretName(backupBucket.Name),
+				Namespace: seedGardenNamespace.Name,
+				Labels:    map[string]string{testID: testRunID},
+			},
+			Data: map[string][]byte{
+				"baz": []byte("dash"),
+			},
+		}
+
+		Expect(testClient.Create(ctx, seedGeneratedSecret)).To(Succeed())
+		log.Info("Created generated Secret in the Seed for test", "secret", client.ObjectKeyFromObject(seedGeneratedSecret))
+
+		DeferCleanup(func() {
+			By("Delete generated Secret in the Seed")
+			Expect(testClient.Delete(ctx, seedGeneratedSecret)).To(Succeed())
+		})
```

Comment:
Why is this done here? Shouldn't this be part of the `backupBucketReady` function? At least you should add a comment that this is usually done by the extension controller which is not running in this test (hence we have do mimic its behaviour)